fastplotlib/widgets/image.py

```diff
@@ -874,15 +833,21 @@ def set_data(
                     f"does not equal current data ndim {current_array.ndim}"
                 )
 
+            # Computes the number of scrollable dims and also validates new_array
+            new_scrollable_dims = self._get_n_scrollable_dims(new_array, self.rgb[i])
+
+            if self.n_scrollable_dims[i] != new_scrollable_dims:
+                raise ValueError(f"Scrollable dims do not match")
```

Comment:
```suggestion
                raise ValueError(f"number of dimensions of data arrays must match number of dimensions of existing data arrays")
```

I think this is what you mean right? because we can't create new sliders etc.